app/helpers/brexit_checker_helper.rb

```diff
@@ -8,14 +8,16 @@ def format_criteria_list(criteria)
   def format_action_audiences(actions)
     action_groups = actions.group_by(&:audience)
 
-    action_groups.map do |key, action_group|
+    action_groups = action_groups.map do |key, action_group|
       {
         heading: I18n.t("brexit_checker.results.audiences.#{key}.heading"),
         actions: action_group.sort_by.with_index do |action, index|
           [-action.priority, index]
         end,
       }
     end
+
+    action_groups.reverse
```

Comment:
Maybe to be certain about the order of the keys, you could use ```sort_by```? https://apidock.com/ruby/Enumerable/sort_by

Reply:
Thanks for running through jos, partition is a much better idea!